packages/content-conformance/src/rules/__tests__/ensure-valid-link-format.test.ts

```diff
@@ -0,0 +1,157 @@
+import { testRule } from '../../test/utils'
+
+import ensureValidLinkFormat from '../ensure-valid-link-format'
+
+describe('ensure-valid-link-format', () => {
+  test('does nothing for valid developer links', () => {
+    testRule(ensureValidLinkFormat, [
+      {
+        fixture: `[I am a valid link](/waypoint/docs/foo/bar)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid link](/boundary/docs/foo/bar)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid tutorial link](/boundary/tutorials/foo/bar)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid collection link](/boundary/tutorials/foo)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid WAF collection link](/well-architected-framework/foo)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid onboarding collection link](/onboarding/foo)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid WAF tutorial link](/well-architected-framework/foo/bar)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid onboarding tutorial link](/onboarding/foo/bar)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid link to an .io domain](https://waypointproject.io/)
+        [I am a valid link to another .io domain](https://terraform.io)`,
+        messages: [],
+      },
+    ])
```

Comment:
🎨 Optional: some additional test cases just to be sure they're covered

```suggestion
      },
      {
        fixture: `[I am a valid link to an .io domain](https://waypointproject.io/not/a/docs/path/)
        [I am a valid link to another .io domain](https://terraform.io/not/a/docs/path)`,
        messages: [],
      },
      {
        fixture: `[I am a valid external link](https://github.com/hashicorp/web-platform-packages)`,
        messages: [],
      },
    ])
```